src/Components/UI/videoformmodal/VideoFormModal.jsx

```diff
@@ -1,117 +1,205 @@
 import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
+import {
+  Formik,
+  Form,
+  Field,
+  ErrorMessage,
+} from 'formik';
+import { useDispatch } from 'react-redux';
 import MyButton from '../button/MyButton';
-import Input from '../input/Input';
+import { createMovie } from '../../../Services/Handlers/AsyncActionsHendlers';
+import { useFormReuquestSelector } from '../../../Services/Selectors/MoviesSelectors';
+import { setFormRequest } from '../../../Services/Actions/MoviesActions';
 
 const emptyVideo = {
   title: '',
+  voteAverage: undefined,
+  releaseDate: '',
   posterPath: '',
-  year: '',
-  genre: ['horro', 'drama'],
-  rating: '',
+  overview: '',
+  genres: ['horro', 'drama'],
   runtime: '',
-  description: '',
 };
 
-function VideoForm({ editVideo, onSubmit, hideModal }) {
+const isValidUrl = (url) => {
+  let correctUrl;
+  try {
+    correctUrl = new URL(url);
+  } catch (e) {
+    return false;
+  }
+  return correctUrl;
+};
+
+function isIsoDate(str) {
+  if (!/\d{4}-\d{2}-\d{2}/.test(str)) return false;
+  return true;
+}
+
+const validation = (values) => {
+  const errors = {};
+  if (!values.title) {
+    errors.title = 'Required';
+  }
+
+  if (!isIsoDate(values.releaseDate)) {
+    errors.releaseDate = 'Incorrect date format. Date must be in YYYY-mm-dd format';
+  }
+
+  if (!values.posterPath) {
+    errors.posterPath = 'Required';
+  } else if (!isValidUrl(values.posterPath)) {
+    errors.posterPath = 'URL is not correct';
+  }
+
+  if (!values.runtime) {
+    errors.runtime = 'Required';
+  }
+
+  if (!values.overview) {
+    errors.overview = 'Required';
+  }
+
+  if (!values.genres) {
+    errors.genres = 'Required';
+  }
+  return errors;
+};
+
+function VideoForm({ editVideo, hideModal }) {
   const [video, setVideo] = useState(emptyVideo);
```

Comment:
I am not sure you need that video state. you are setting editVideo from props there and then pass to component... why not pass editVideo directly without setting it into video?